scipy/stats/_stats_py.py

```diff
@@ -3836,7 +3836,7 @@ def _alexandergovern_input_validation(samples, nan_policy):
                                                                  "pvalue"))
 
 
-class AlexanderGovernConstantInputWarning(RuntimeWarning):
+class AlexanderGovernConstantInputWarning(stats.ConstantInputWarning):
```

Comment:
Maybe simplify it further and emit a ConstantInputWarning, here and elsewhere?

Reply:
I left it that way based on what @rgommers wrote [here](https://github.com/scipy/scipy/issues/13462#issuecomment-770392420): leave the existing test-specific warnings, but only make a more general superclass warning.

Reply:
I wasn't actually advocating for that, what @ev-br says it's better. I just meant that it's acceptable to keep this private if you'd really want function-specific warnings. But to me it doesn't have any value, just code bloat.

Reply:
I don't disagree. I just went with what was written there.
I eliminated `AlexanderGovernConstantInputWarning` because it wasn't public.
I eliminated `BootstrapDegenerateDistributionWarning` because I'm comfortable doing so, as it is very recent and probably nobody uses it.
I left the rest alone because they have been public for a while and I don't fully understand the original rationale for making them separate in the first place, given that they considered a single warning for all of them at the time.
Let me know if you want those removed from the `stats` namespace without warning, though.

Reply:
> I left the rest alone because they have been public for a while

Fair enough. Perhaps they're in use somewhere. It's not great design, but I agree with the current state of this PR - no need to rip it out without a deprecation. How about leaving it as is and opening an issue to deprecate the rest? Then it can be done later, perhaps as an intro task for a newer contributor.

Reply:
OK. For reference, this would entail:
* `warnings.warn(msg, DeprecationWarning)`, where `msg` is something like:

> This function currently emits a subclass of `ConstantInputWarning`, `F_onewayConstantInputWarning`. As of SciPy 1.9.0, `F_onewayConstantInputWarning` is deprecated in favor of its base class and will be removed in SciPy 1.11.0. 

* Adding `assert_warns(DeprecationWarning)` in all the tests that would start to fail
* Adding a notice in the release notes
* Adding a `.. warning::` or `.. note::` to the "Warns" section of the documentation with a message similar to the one above

Reply:
I'll take care of this to get the deprecation in 1.9.0.